oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Configuration.java

```diff
@@ -417,4 +418,11 @@
         name = "Enable Full GC Persistent Audit Logging",
         description = "This parameter will enable/disable the saving of deleted document IDs and properties during FullGC into a persistent storage, e.g Mongo collection")
     boolean fullGCAuditLoggingEnabled() default false;
+
+    @AttributeDefinition(
+            name = "Avoid Exclusive Merge lock",
+            description = "Boolean value indicating whether we need to avoid the exclusive merge lock while " +
+                    "merging the changes in case of a conflict. The Default value is " + DEFAULT_AVOID_MERGE_LOCK +
+                    " Note that this value can be overridden via framework property 'oak.documentstore.avoidMergeLock'")
+    boolean avoidMergeLock() default DEFAULT_AVOID_MERGE_LOCK;
```

Comment:
doesn't have to be everywhere else, but just wondered if the config param should contain the word `exclusive` - just to be more in line with the code, so eg

```suggestion
    boolean avoidExclusiveMergeLock() default DEFAULT_AVOID_MERGE_LOCK;
```

Reply:
I think this change makes sense to be more informative. Also in the property if possible:
`oak.documentstore.avoidExclusiveMergeLock`.

Reply:
Done